ParcelDistributionCenter/ParcelDistributionCenter.Web/Controllers/PackagesController.cs

```diff
@@ -97,5 +98,26 @@ public ActionResult Index()
         {
             return View();
         }
-    }
+        public ActionResult FindByPackageID(int packageID)
+        {
+            try
+            {             
+                var model = _packageHandler.FindPackageByNumber(packageID);
+
+                if (model == null) 
+                {
+                  return  RedirectToAction("InsertPackageID", "Packages"); 
+                }
+                return View(model);
+            }
+            catch
+            {
+                return RedirectToAction("InsertPackageID", "Packages");
+            }
+        }
```

Comment:
Nie do końca rozumiem idee try / catch. 

Może to odwrócić? Jeżeli model !=null  to return View(model)
A w przeciwnym razie Redirect( "InsertPackageID", "Packages")